### PR TITLE
6253-submit-reporting-report

### PR DESCRIPTION
**Pull request by bthaile**
- added call to submit initial report to make contract call to do initial report, used for DR and open reporting initial reporter. 
- integrated submit initial report into reporting form submit. 
- added helper class to calculate payout numerators
- added test for helper
- fixed setting convenience properties for isScalar, isBinary, ....

https://app.clubhouse.io/augur/story/6253/reporting-form-submit-button

### CODE REVIEW

**Comment by sharkcrayon** [comment]
Related CH story: https://app.clubhouse.io/augur/story/6253/reporting-form-submit-button

File: src/modules/reporting/actions/get-payout-numerators.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (stephensprinkle) [inline src/modules/reporting/actions/get-payout-numerators.js]: This is actually a selector, not an action since it’s not making any calls or updating state.

Move this to the selectors directory + also add memoization.

File: src/modules/reporting/components/reporting-report/reporting-report.jsx [hunk 2]
```diff
@@ -95,7 +96,11 @@ export default class ReportingReport extends Component {
             >Report
             </button>
             { s.currentStep === 1 &&
-              <button className={FormStyles.Form__submit}>Submit</button>
+              <button
+                className={FormStyles.Form__submit}
+                onClick={e => p.submitInitialReport(p.market.id, p.selectedOutcome)}
```
Comment (stephensprinkle) [inline src/modules/reporting/components/reporting-report/reporting-report.jsx]: No need for `e` here, can just do `() => `

File: src/modules/reporting/containers/reporting-report.js [hunk 3]
```diff
@@ -2,16 +2,18 @@ import { connect } from 'react-redux'
...
 
 const mapStateToProps = state => ({
   isLogged: state.isLogged,
   market
 })
 
-// const mapDispatchToProps = dispatch => ({
-// })
+const mapDispatchToProps = dispatch => ({
+  submitInitialReport: (marketId, outcomeValue, invalid) => { dispatch(submitInitialReport(marketId, outcomeValue, invalid)) }
```
Comment (stephensprinkle) [inline src/modules/reporting/containers/reporting-report.js]: Can shorten this to 
```
(marketId, outcomeValue, invalid) => dispatch(submitInitialReport(marketId, outcomeValue, invalid))
```
